lib/services/ios-project-service.ts

```diff
@@ -300,9 +304,62 @@ export class IOSProjectService extends projectServiceBaseLib.PlatformProjectServ
 
 			let xcodeBuildVersion = this.getXcodeVersion();
 			if (helpers.versionCompare(xcodeBuildVersion, "8.0") >= 0) {
-				let teamId = this.getDevelopmentTeam();
-				if (teamId) {
-					args = args.concat("DEVELOPMENT_TEAM=" + teamId);
+				// TRICKY: I am not sure why we totally disregard the buildConfig parameter here.
+				buildConfig = buildConfig || {};
+
+				if (this.$options.teamId) {
+					buildConfig.teamIdentifier = this.$options.teamId;
+				} else {
+					buildConfig = this.readXCConfigSigning();
+					if (!buildConfig.codeSignIdentity && !buildConfig.mobileProvisionIdentifier && !buildConfig.teamIdentifier) {
+						buildConfig = this.readBuildConfigFromPlatforms();
+					}
+				}
+
+				let signingStyle: XcodeSigningStyle;
+				if (buildConfig.codeSignIdentity || buildConfig.mobileProvisionIdentifier) {
+					signingStyle = "Manual";
+				} else if (buildConfig.teamIdentifier) {
+					signingStyle = "Automatic";
+				} else {
+					let signingStyles = [
+						"Manual - Select existing provisioning profile for use",
+						"Automatic - Select Team ID for signing and let Xcode select managed provisioning profile"
+					];
+					let signingStyleIndex = signingStyles.indexOf(this.$prompter.promptForChoice("Select codesiging style", signingStyles).wait());
```

Comment:
What will happen in non-interactive terminal? The prompter will wait for input, but there will be no input, so maybe the execution will hang here.